src/epggrab/otamux.c

```diff
@@ -673,13 +673,26 @@ epggrab_ota_kick_cb ( void *p )
                                 SUBSCRIPTION_TABLES))) {
     if (r != SM_CODE_NO_ADAPTERS) {
       tvhtrace(LS_EPGGRAB, "subscription failed for %s (result %d)", mm->mm_nicename, r);
-      TAILQ_INSERT_TAIL(&epggrab_ota_pending, om, om_q_link);
-      om->om_q_type = EPGGRAB_OTA_MUX_PENDING;
-      om->om_retry_time = mclk() + mono2sec(60);
-      if (r == SM_CODE_NO_FREE_ADAPTER)
-        net->failed = 1;
-      if (r == SM_CODE_TUNING_FAILED)
+      if( om->om_retry_count < 3 ) {
+        TAILQ_INSERT_TAIL(&epggrab_ota_pending, om, om_q_link);
+        om->om_q_type = EPGGRAB_OTA_MUX_PENDING;
+        if (r == SM_CODE_NO_FREE_ADAPTER) {
+          /* wait for scheduled mux to finish OTA grab */		 
+          om->om_retry_time = mclk() + sec2mono(60);
+	  om->om_retry_count = 0;
+	}
+	else {
+          /* cannot tune - increase retry count and try again later */		 
+          om->om_retry_time = mclk() + sec2mono(30);
+	  ++(om->om_retry_count);
+	}
         net->failed = 1;
+      }
+      else {
+        tvhtrace(LS_EPGGRAB, "too many failed attempts to tune %s (subscribe)", mm->mm_nicename);
+        net->fatal = 1;
+	om->om_retry_count = 0;
```

Comment:
Intendation is wrong here

Reply:
yes - i'll need to fix my tabbing - thanks for spotting this